novawallet/Common/Services/ChainRegistry/ConnectionPool/ConnectionFactory.swift

```diff
@@ -162,7 +162,11 @@ extension ConnectionFactory: ConnectionFactoryProtocol {
     }
 
     private func extractNodeUrls(from chain: ChainModel, schema: String) -> [ConnectionTLSSupport] {
-        let filteredNodes = chain.nodes.filter { $0.url.hasPrefix(schema) }
+        let filteredNodes = if case let .manual(selectedNode) = chain.connectionMode {
```

Comment:
I think we need to return empty set if schema of the selected node not matching the requested schema.